### PR TITLE
fix: circular imports

### PR DESCRIPTION
**Pull request by MDavidson17**
Changed import statements and used [typing.TYPE_CHECKING](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING) to resolve circular import errors. 

**note:** [(VIDEO) A good explaination of the circular import error and resolution](https://www.youtube.com/watch?v=B5cjckVzY4g&t=315s&ab_channel=anthonywritescode)

### CODE REVIEW

File: topo_processor/data/data_transformers/data_transformer_imagery_historic.py [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
 
+import topo_processor.stac as stac
 from topo_processor.cog.create_cog import create_cog
-from topo_processor.stac import Asset, Item
 from topo_processor.util import is_tiff, time_in_ms
 
 from .data_transformer import DataTransformer
 
+if TYPE_CHECKING:
+    from topo_processor.stac import Item
```
Comment (amfage) [inline topo_processor/data/data_transformers/data_transformer_imagery_historic.py:17]: Does Asset need to be imported as well as Item?

File: topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py [hunk 2]
```diff
@@ -17,7 +23,7 @@ def is_applicable(self, asset: Asset) -> bool:
         return is_tiff(asset.source_path)
 
     async def load_metadata(self, asset: Asset) -> None:
-        asset.item.add_extension(StacExtensions.projection.value)
+        asset.item.add_extension(stac.StacExtensions.projection.value)
```
Comment (amfage) [inline topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py:26]: Need to import StacExtensions?

**Review by amfage — APPROVED** [review]

**Review by paulfouquet — APPROVED** [review]

**Review by blacha — APPROVED** [review]

**Comment by blacha** [comment]
Does the 

```
from __future__ import annotations
```

help the type checking somehow?

my quick google hasnt really helped me on it

**Comment by MDavidson17** [comment]
> Does the
> 
> ```
> from __future__ import annotations
> ```
> 
> help the type checking somehow?
> 
> my quick google hasnt really helped me on it

Using that import stops us from having to use quotes. 

Without import: 
```
def add_item(self, item: "Item"):
```
with import:
```
def add_item(self, item: Item):
```
He doesn't go into too much detail but he mentions it in the video I linked.

**Pull request merged**